lessons/02-javascript/README.md

```diff
@@ -152,6 +157,8 @@ for (var key in object) {
 }
 ```
 
+Note that the order of iteration is not guarenteed with this syntax and later, if you use libraries which modify the prototype (more on this later) of base types, can have strange behavior.
```

Comment:
As an alternative to Object.keys.... you could introduce:
```node
for (var key in p) {
  if (p.hasOwnProperty(key)) {
    var val = p[key];
  }
}
```
But I've always prefered this first style personally.